src/gtnh/gui/exclusion_frame.py

```diff
@@ -0,0 +1,157 @@
+from tkinter import END, Button, Entry, LabelFrame, Listbox, Scrollbar, StringVar
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+
+class ExclusionFrame(LabelFrame):
+    """Widget managing an exclusion list."""
+
+    def __init__(
+        self, master: Any, frame_name: str, callbacks: Dict[str, Any], width: Optional[int] = None, **kwargs: Any
+    ) -> None:
+        """
+        Constructor of the ExclusionFrame class.
+
+        :param master: the parent widget
+        :param frame_name: the name displayed in the framebox
+        :param callbacks: a dict of callbacks passed to this instance
+        :param kwargs: params to init the parent class
+        """
+        LabelFrame.__init__(self, master, text=frame_name, **kwargs)
+        self.xpadding: int = 0  # todo: tune this
+        self.ypadding: int = 0  # todo: tune this
+        self.btn_add_text: str = "add new exclusion"
+        self.btn_del_text: str = "remove highlighted"
+        self.width: int = width if width is not None else max(len(self.btn_add_text), len(self.btn_del_text))
+        self.listbox: Listbox = Listbox(self, exportselection=False, height=16)
+        self.sv_entry: StringVar = StringVar(value="")
+        self.entry: Entry = Entry(self, textvariable=self.sv_entry)
+        self.btn_add: Button = Button(self, text="add new exclusion", command=self.add)
+        self.btn_del: Button = Button(self, text="remove highlighted", command=self.delete)
+        self.add_callback: Callable[[str], None] = callbacks["add"]
```

Comment:
nit: I generally prefer to go with str Enums for this, avoids typos and lets the IDE do auto complete.

You could also probably move the typing to the callback Dict[] definition if you wanted.  🤷

Reply:
well i extracted the texts a couple of lines before, but i forgot to call the variables :(